### PR TITLE
Events refactor audio

### PR DESCRIPTION
**Pull request by Steelstone3**

### CODE REVIEW

File: src/systems/events/spawn_audio.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+#[allow(dead_code)]
+pub fn spawn_audio(
+    mut spawn_audio_events: EventReader<SpawnAudioEvent>,
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+) {
+    for spawn_audio_event in spawn_audio_events.read() {
+        //If passed entity exists
+        if let Some(mut entity) = commands.get_entity(spawn_audio_event.entity) {
```
Comment (Steelstone3) [inline src/systems/events/spawn_audio.rs:20]: Not sure why this doesn't work...

File: src/systems/music/exploration_music.rs [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
-use crate::resources::music::Music;
-
 pub fn play_exploration_music(
     mut commands: Commands,
-    asset_server: Res<AssetServer>,
     music: Res<Music>,
+    asset_server: Res<AssetServer>,
+    // mut spawn_audio_event: EventWriter<SpawnAudioEvent>,
 ) {
+    // let entity = commands.spawn_empty().id();
```
Comment (Steelstone3) [inline src/systems/music/exploration_music.rs:14]: Or this also not working

File: src/systems/events/spawn_audio.rs (hunk 1 above)
Comment (MatthewGriffin) [reply src/systems/events/spawn_audio.rs:20]: you need to register the event in game_events_plugins

File: src/systems/music/exploration_music.rs (hunk 2 above)
Comment (MatthewGriffin) [reply src/systems/music/exploration_music.rs:14]: probably  need to expand spawn audio to let it handle scenarios where there is no entity to attach to

**Pull request merged**